### PR TITLE
fix: do not crash findComponent if ref is pointing to html element

### PR DESCRIPTION
**Pull request by xanf**
This is first part of fix related to #716 

After certain consideration, I've decided that expected behavior should be not throwing error, but returning empty wrapper instead.
My reasoning is based on three points:

1. Passing `ref` to `findComponent` is perfectly valid syntax. So if we're looking for component with specific ref and it does not exists - why should we complain about HTML element with same ref? We simply should not care
2. (Real world example from gitlab codebase). Consider having `<component ref="hello" :is="foo">` where `foo` under certain conditions is resolved either to `<div>` or to component (let's say `<Note>`). Consider having table syntax test, where we pass certain props... and if we throw, it definitely will look ugly and wrong instead of just checking for `.exists()`
3. `findComponent` does not have `throw` behavior and I'm resisting adding one (keeping mental mode simple - `getComponent` throws, `findComponent` - does not thow)

### CODE REVIEW

File: src/vueWrapper.ts [hunk 1]
```diff
@@ -159,8 +159,11 @@ export class VueWrapper<T extends ComponentPublicInstance>
   ): VueWrapper<T> {
     if (typeof selector === 'object' && 'ref' in selector) {
       const result = this.vm.$refs[selector.ref]
-      if (result) {
+      console.log(result)
```
Comment (afontcu) [inline src/vueWrapper.ts:162]:
```suggestion
```

**Review by afontcu — APPROVED** [review]
Looks good to me! 🚀  @lmiller1990 since you expressed some concerns in https://github.com/vuejs/vue-test-utils-next/issues/716, do you believe this makes sense?

I also suggested throwing, but I agree that the differente between `find` and `get` is easier to grasp if the former never throws – even on edge cases.

File: src/vueWrapper.ts (hunk 1 above)
Comment (xanf) [reply src/vueWrapper.ts]: @afontcu Thank you! One day I will remember to have another SPECIFIC look for console.logs and debugger :)

**Comment by lmiller1990** [comment]
Seems fine! This is the V1 behavior, right? 

I really regret the decision to support both `findComponent` AND `getComponent`. The differences are so subtle, and it's really hard to remember with different libraries using `get`, `find`, and `query` all slightly differently.

**Comment by xanf** [comment]
> My concern is that it'll be confusing to return an empty wrapper that doesn't actually do anything. Does this really provide any value other than some fairly specific cases? Eg, even if is returns a wrapper, you cannot do very much except assert exists (and there should be another way to write that same test). Should we not just return undefined (since there is no component with the ref you are querying for? This seems like the more logical behavior to me.

Well, if we're speaking in scope of this PR - it definitely should be an empty wrapper, just because this is what `findComponent` (and `find`) returns when no result is found, no matter which selector is used, and keeping this behavior consistent is pretty important IMO

If you're speaking about returning undefined instead of empty wrapper "in general" for any find operation - I do not have strong preference towards any of the approaches and both makes sense to me, but I know that at least some of my colleagues like "JQuery-like" behavior of `find / findComponent` - when you could always be sure you will have wrapper as a result even if it is empty

**Comment by afontcu** [comment]
I think I also lean towards returning an empty Wrapper – as we've already pointed out, we should teach users to use `get(Component)` if they are certain an item is there (let's bear in mind we added `find(Component)` to allow people to make sure a component/node is present!)

**Comment by lmiller1990** [comment]
Ok, I will trust you guys on this one 👍
